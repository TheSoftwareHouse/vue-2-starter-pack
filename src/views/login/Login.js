export default {
  name: "Login",
  methods: {
    handleSubmit() {
      this.$emit("submit");
    },
  },
};
