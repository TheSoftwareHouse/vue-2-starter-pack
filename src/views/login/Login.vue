<template>
  <div class="login">
    <h1>This is login page</h1>
    <form @submit.prevent="handleSubmit">
      <div>
        <label>
          username:
          <input name="username" />
        </label>
      </div>
      <div>
        <label>
          password:
          <input name="password" type="password" />
        </label>
      </div>
      <button type="submit">submit</button>
    </form>
  </div>
</template>

<script src="./Login.js"></script>
