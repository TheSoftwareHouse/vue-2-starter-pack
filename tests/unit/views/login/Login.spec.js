import { mount } from "@vue/test-utils";

import Login from "@/views/login/Login.vue";

describe("Login view", () => {
  it("should render correctly", () => {
    const wrapper = mount(Login);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should emit submit event", () => {
    const div = document.createElement("div");
    document.body.appendChild(div);

    const wrapper = mount(Login, { attachTo: div });

    wrapper.find("button").trigger("click");

    expect(wrapper.emitted().submit).toBeTruthy();
  });
});
