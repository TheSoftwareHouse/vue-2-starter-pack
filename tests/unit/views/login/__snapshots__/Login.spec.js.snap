// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login view should render correctly 1`] = `
<div class="login">
  <h1>This is login page</h1>
  <form>
    <div><label>
        username:
        <input name="username"></label></div>
    <div><label>
        password:
        <input name="password" type="password"></label></div> <button type="submit">submit</button>
  </form>
</div>
`;
